ui/src/app/views/resource-card/component.spec.ts

```diff
@@ -0,0 +1,132 @@
+        ResourceCardComponent,
+        ConfirmDeleteModalComponent,
+        RenameModalComponent,
+        CohortEditModalComponent,
+      ],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      providers: [
+        {provide: CohortsService, useValue: new CohortsServiceStub()},
+        {provide: SignInService, useValue: new SignInServiceStub()},
+        {provide: WorkspacesService, useValue: new WorkspacesServiceStub()},
+      ]
+    }).compileComponents()
+      .then(() => {
+      fixture = TestBed.createComponent(ResourceCardWrapperComponent);
+      component = fixture.debugElement.children[0].componentInstance;
+
+      fixture.detectChanges();
+    });
+  }));
+
+  it('should render', fakeAsync(() => {
+    expect(component).toBeTruthy();
+  }));
+
+  // it should render cohort if resource is cohort
+  it('should render cohort if resource is cohort', fakeAsync(() => {
+    expect(fixture.debugElement.query(By.css('.cohort-card'))).toBeTruthy();
+    expect(fixture.debugElement.query(By.css('#cohort-decoration'))).toBeTruthy();
+  }));
+
+  // it should render notebook if resource is notebook
+  it('should render notebook if resource is notebook', fakeAsync(() => {
+    setNotebookResource(fixture, component);
+    updateAndTick(fixture);
+    expect(fixture.debugElement.query(By.css('.notebook-card'))).toBeTruthy();
+  }));
+
+  // Actual actions are tested in cohort list/notebook list specs...
+  // CURRENTLY those tests don't technically test these actions, but will once
+  // planned refactoring takes place -US
+  it('should have correct options in menu when it is a cohort', fakeAsync(() => {
+    const de = fixture.debugElement;
+    simulateClick(fixture, de.query(By.css('.dropdown-toggle')));
+    const actionItems = de.queryAll(By.css('.action-item'))
+      .map((item) => item.nativeElement.innerText.trim());
+    ['Clone', 'Edit', 'Review', 'Delete'].forEach((action) => {
+      expect(actionItems).toContain(action);
+    });
+  }));
+
+  it('should have correct options in menu when it is a notebook', fakeAsync(() => {
+    setNotebookResource(fixture, component);
+    const de = fixture.debugElement;
+    simulateClick(fixture, de.query(By.css('.dropdown-toggle')));
+    const actionItems = de.queryAll(By.css('.action-item'))
+      .map((item) => item.nativeElement.innerText.trim());
+    ['Rename', 'Clone', 'Delete'].forEach((action) => {
+      expect(actionItems).toContain(action);
+    });
+  }));
+});
+
+function setNotebookResource(fixture, component) {
+  component.resourceCard = {
+    workspaceId: 1,
+    workspaceNamespace: 'defaultNamespace',
+    workspaceFirecloudName: 'defaultFirecloudName',
+    permission: 'Owner',
+    modifiedTime: Date.now().toString(),
+    notebook: {
+      'name': 'mockFile.ipynb',
+      'path': 'gs://bucket/notebooks/mockFile.ipynb',
+      'lastModifiedTime': 100
+    },
+  };
+  // call OnInit here because that's where ResourceCard reads the resourceType
+  component.ngOnInit();
```

Comment:
FYI: I recalled having solved this issue during review, but I just happened upon the code. The downside is that every test will need to call this initially (or *maybe* you could call it once in beforeEach, and optionally a second time here...). The upside is that you don't need to hand-replicate the angular lifecycle and injection:
https://github.com/all-of-us/workbench/blob/876114ab08ff32f4c197874882b1bcb49c102bcd/ui/src/app/views/workspace-edit/component.spec.ts#L34

Reply:
hmmm, so not sure what you want me to change here. The example you gave is not exactly the same. I am still foggy from this cold, so sorry if I'm being dim. Leaving as is for now.